### DISCUSSION TITLE
Semantic tokens and custom CSS selectors

### DISCUSSION THREAD

**Discussion by khinshankhan** [chakra-ui/chakra-ui#5988] [General]
After reading up on semantic tokens and using them a bit, something from the documentation caught my eye (https://chakra-ui.com/docs/styled-system/features/semantic-tokens):

> The condition can be one of [chakra pseudo selectors](https://chakra-ui.com/docs/styled-system/features/style-props#pseudo) or a custom CSS selector.

The pseudo selectors mostly make sense with eg `_dark`, however, I haven't been able to find any documentation around the custom selectors. Not even sure if custom selector means what I think it means: we can define our own css variable to interpret? 

So this is a roundabout request for elaboration and hopefully examples.

**Comment by jsvossen** [comment]
Even though this is an old post, I wanted to share what I learned in case anyone else stumbles across this: they literally mean any custom css selector, e.g. `.my-class`. So the color token "bodyText" might look like:
```
colors: {
  bodyText: {
    default: "black",
    _dark: "white",
   ["&.my-class"]: "red"
  }
}
```
In my case, I'm trying to add a high contrast mode. I add [data-high-contrast=true] to the page body and set the tokens up like this:
```
const _hc = "[data-high-contrast=true] &"
const semanticTokens = {
  colors: {
    mainBackground: {
      default: "white",
      _dark: "gray.800",
      [_hc]: "black"
    }
  }
}
```

**Reply by prtcl** [reply]
Apologies for the old thread bump but I'm trying to do something similar. But in 2.7 it looks like placing any string/selector in the semantic token object just breaks the resolution of CSS selector to var mapping (the CSS var it outputs has an invalid name, and the components can't resolve my semantic token). Is there anything special you had to do, like define your own pseudo selectors somewhere?

**Reply by jsvossen** [reply]
Dang, I just noticed this too. I opened a new bug, so hopefully we can get an answer: https://github.com/chakra-ui/chakra-ui/issues/7744
